### ISSUE TITLE
typescript error

### ISSUE THREAD

**Issue by Console45** [tannerlinsley/react-query#870]
TypeScript error in C:/Users/Cosmos/OneDrive/Desktop/react-infinite-scroll/node_modules/react-query/types/core/index.d.ts(5,1):
Declaration or statement expected.  TS1128

      4 | export * from './types';
  > 5 | export type { Query } from './query';
        | ^
    6 | export type { QueryCache } from './queryCache';
    7 | export type { ConsoleObject } from './utils';
    8 | 

cant get passed this. when i import the lib i throws this error immediately

**Comment by boschni** [comment]
Hi @Console45! Are you running a TypeScript version below 3.8?

**Comment by Console45** [comment]
No my version is 3.9.5

**Comment by michaelwooley** [comment]
@Console45 here is my less than ideal workaround:

```ts
// @ts-ignore
import { useQuery } from 'react-query/index';
import { QueryConfig, QueryResult } from 'react-query/types/core/types.d';
```

**Comment by boschni** [comment]
Which `react-query` version are you running? Would it be possible to provide a playground example?

**Comment by michaelwooley** [comment]
@boschni  thanks for your response. I misinterpreted your earlier comment to mean that the version ought to be _below_ 3.8. Today I started using typescript 3.9.7 and all is well. I also noticed the "fine print" in the docs about this. My bad.

I was using `react-query-2.5.7.-tsnext.1` with `typescript-3.7.5`.

[3.7.5 is currently the default version](https://github.com/facebook/create-react-app/blob/v3.4.1/packages/cra-template-typescript/template.json) with the typescript create-react-app template. Updating the typescript version doesn't cause any issues for my case.

I could make an example of what not to do if you like/think it would help.
